src/plugins/intel_npu/src/plugin/src/plugin.cpp

```diff
@@ -539,6 +489,115 @@ Plugin::Plugin()
         {ov::intel_npu::batch_mode.name(), {false, ov::PropertyMutability::RW, [](const Config& config) {
                                                 return config.getString<BATCH_MODE>();
                                             }}}};
+    return temp_prop;
+}
+
+/**
+ * @brief init npu-plugin normally
+ * @details  this is original npu-plugin constructor
+ */
+void Plugin::init(){
+    // TODO: generation of available backends list can be done during execution of CMake scripts
+    std::vector<AvailableBackends> backendRegistry;
+
+#if defined(OPENVINO_STATIC_LIBRARY)
+    backendRegistry.push_back(AvailableBackends::LEVEL_ZERO);
+#else
+#    if defined(ENABLE_IMD_BACKEND)
+    if (const auto* envVar = std::getenv("IE_NPU_USE_IMD_BACKEND")) {
+        if (envVarStrToBool("IE_NPU_USE_IMD_BACKEND", envVar)) {
+            backendRegistry.push_back(AvailableBackends::IMD);
+        }
+    }
+#    endif
+
+#    if defined(_WIN32) || defined(_WIN64) || (defined(__linux__) && defined(__x86_64__))
+    backendRegistry.push_back(AvailableBackends::LEVEL_ZERO);
+#    endif
+#endif
+
+    OV_ITT_TASK_CHAIN(PLUGIN, itt::domains::NPUPlugin, "Plugin::Plugin", "NPUBackends");
+    _backends = std::make_shared<NPUBackends>(backendRegistry, _globalConfig);
+    OV_ITT_TASK_NEXT(PLUGIN, "registerOptions");
+    _backends->registerOptions(*_options);
+
+    OV_ITT_TASK_NEXT(PLUGIN, "Metrics");
+    _metrics = std::make_unique<Metrics>(_backends);
+
+    // parse again env_variables after backend is initialized to get backend proprieties
+    _globalConfig.parseEnvVars();
+
+    // Map from name to function {Config -> ov::Any}
+    // Note that some properties are RW before network is loaded, and become RO after network is loaded
+    // this is for dry on execution.
+    _properties = Generate_supportedPropert();
+    _properties.insert(
+        // OV Internals
+        // =========
+        {ov::internal::supported_properties.name(),
+         {false,
+          ov::PropertyMutability::RO,
+          [&](const Config&) {
+              return _metrics->GetInternalSupportedProperties();
```

Comment:
this part can be replaced by `static std::vector<ov::PropertyName> Metrics_internalSupportedProperties = {ov::internal::caching_properties.name()};`

 is not it?